### PR TITLE
C# symbol matcher changes and test to support tuples and ref

### PR DESCRIPTION
**Pull request by ivanbasov**
**Customer scenario**

Support C# 7.0 features in EnC:
 - tests for tuples  
- fix to support ref

**Bugs this fixes:**

Works towards to #17891

**Workarounds, if any**

**Risk**

Low

**Performance impact**

Low

**Is this a regression from a previous update?**

**Root cause analysis:**

**How was the bug found?**

Planned

### CODE REVIEW

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.cs [hunk 1]
```diff
@@ -519,5 +519,432 @@ static void F()
...
+                default(EmitContext),
+                compilation0.SourceAssembly,
+                default(EmitContext),
+                null);
+
+            var member = compilation1.GetMember<FieldSymbol>("Vector.Coordinates");
+            var other = matcher.MapDefinition(member);
+            // Types must match because just an element name was changed.
+            Assert.NotNull(other);
+        }
```
Comment (jcouv) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.cs]: Nit: Extra empty line below

**Comment by tmat** [comment]
:shipit:

File: src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.cs (hunk 1 above)
Comment (ivanbasov) [inline src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/SymbolMatcherTests.cs]: Thanks! Fixed

**Comment by ivanbasov** [comment]
Tagging @MattGertz for Ask mode approval

**Comment by jinujoseph** [comment]
LGTM

**Pull request merged**